packages/grid/_modules_/grid/utils/warning.ts

```diff
@@ -0,0 +1,26 @@
+export const buildWarning = (message: string | string[]) => {
+  let alreadyWarned = false;
+
+  const cleanMessage = Array.isArray(message) ? message.join('\n') : message;
+
+  return () => {
+    if (!alreadyWarned) {
+      alreadyWarned = true;
+      console.warn(cleanMessage);
+    }
+  };
+};
+
+export const wrapWithWarningOnCall = <Args extends any[], R extends any>(
+  method: (...args: Args) => R,
+  message: string | string[],
+) => {
+  if (process.env.NODE_ENV === 'production') {
+    return method;
+  }
+
+  return (...args: Args) => {
+    buildWarning(message);
```

Comment:
Is it dispatching only one warning? I thought the build warning should be outside of the returned method, as follow
 
```js
const warning = buildWarning(message);
return (...args: Args) => {
    warning();
    ...
}
```

Reply:
Oh you are right...